src/Repository/FileRepository.php

```diff
@@ -25,4 +27,23 @@ public function __construct(ManagerRegistry $registry)
     {
         parent::__construct($registry, File::class);
     }
+
+    /**
+     * Query for getting filename and fileapth.
+     *
+     * @param array $fileIds List of fileIds to be queried on.
+     *
+     * @return array
+     */
+    public function getFileNameAndPath(array $fileIds) : array
```

Comment:
We don't really need this function? The final result and not needed for something like before. It is just as fast to go trough the about, and then use the functions in the entity instead.

Reply:
I think we still need it, because we give the list of File ids to the message handler. We need to retrieve the file entities, we can just use findBy(criteria), then we need to filter out to get filename and filepath

Reply:
You're right, we need it. It's not always all files in a dataset.